lib/galaxy/managers/datasets.py

```diff
@@ -758,7 +758,9 @@ def serialize_rerunnable(self, item, key, **context):
         """
         dataset = item
         if dataset.creating_job:
-            tool = self.app.toolbox.get_tool(dataset.creating_job.tool_id, dataset.creating_job.tool_version)
+            tool = self.app.toolbox.tool_for_job(
+                dataset.creating_job, exact=False, check_access=True, user=context.get("user")
```

Comment:
I guess the check access check didn't exist until this commit but it feels like it should be checked? We shouldn't be revealing parameter details and such that would be rerunnable if the user doesn't access right? Or is that done for the job - and if so want to add a comment here.

Reply:
I am not sure I understand ? So rerunning didn't work at all because we we're considering user defined and dynamic tools in `self.app.toolbox.get_tool`. We do that now (tool_for_job is a utility method that looks at the job and builds the tool from job.dynamic_tool) and we check access by default (see the function signature). The only time we don't do that is in the job handler ... if the job is running surely there's no point in checking.

All existing code that builds user defined tools calls ensure_can_use_unprivileged_tool (see the DynamicToolManager code).

Reply:
IOW anything to do with user defined tools requires permission to use them, I suppose maybe I don't understand what you had in mind for a comment ?

Reply:
I'm so so so sorry to waste your time - I read that as disabling check access. Ugh - sorry for the noise.

Reply:
Also I suppose I could have clarified this with extra tests ... that's still on my todo list.